File: pubnub-gson/src/main/java/com/pubnub/api/endpoints/objects_api/channel/GetChannelMetadata.java
package com.pubnub.api.endpoints.objects_api.channel;

import com.pubnub.api.endpoints.BuilderSteps;
import com.pubnub.api.endpoints.Endpoint;
import com.pubnub.api.models.consumer.objects_api.channel.PNGetChannelMetadataResult;

public interface GetChannelMetadata extends Endpoint<PNGetChannelMetadataResult> {
    GetChannelMetadata includeCustom(boolean includeCustom);

    interface Builder extends BuilderSteps.ChannelStep<GetChannelMetadata> {
        @Override
        GetChannelMetadata channel(String channel);
    }
}

File: pubnub-kotlin/pubnub-kotlin-api/src/jvmMain/kotlin/com/pubnub/api/endpoints/objects/channel/GetChannelMetadata.kt
package com.pubnub.api.endpoints.objects.channel

import com.pubnub.api.Endpoint
import com.pubnub.api.models.consumer.objects.channel.PNChannelMetadataResult

/**
 * @see [PubNub.getChannelMetadata]
 */
interface GetChannelMetadata : com.pubnub.kmp.endpoints.objects.channel.GetChannelMetadata, Endpoint<PNChannelMetadataResult>
